### ISSUE TITLE
Alphageometry seems not ablt to solve basic geometry question

### ISSUE THREAD

**Issue by maokaiii** [google-deepmind/alphageometry#99]
Let A B C be a triangle. ∠A = 50° , ∠B = 70°. Find ∠C.

Translated problem:
`a b = segment a b; c = s_angle b a c -50, s_angle a b c 80; d = eq_triangle b c ? eqangle c b c a  b d b c`

**Comment by maokaiii** [comment]
Or even this one:

```a b = segment a b; c = on_circle a b; d = on_line a b; e = on_aline b a a c d ? cong a d a e```

**Comment by maokaiii** [comment]
> Or even this one:
> 
> `a b c = iso_triangle a b c ? eqangle a b b c a c c b`

this seems to be the problem of directed angle, the order of side matters

**Comment by maokaiii** [comment]
It can't even do this?
```
a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c a c b
```

**Comment by XiaoYang66** [comment]
Hello,  can you please tell me how do you know how to interpret the natural language into the language by dd+ar?

**Comment by maokaiii** [comment]
Just by reading IMO questions and compare it to imo txt file. Also read the nature paper and def.txt

**Comment by XiaoYang66** [comment]
Ok, thank you very much. That help me a lot. Actually, I thought I might ignore some important file. But actually, the authors did not explain the detailed implementation just like the DD authors did([A Deductive Database Approach to Automated Geometry Theorem Proving and Discovering](https://link.springer.com/article/10.1023/A:1006171315513))

I also find some simple question can not be solved by the DD+AR implementation, such as:
`a b c d = perp a b c d; e f = perp e f c d, ncoll a b e ? para a b e f` or 
`a b c = iso_triangle a b c ? eqdistance a b c a`

**Comment by maokaiii** [comment]
I think it should be 
`a b c = iso_triangle a b c ? cong a b a c`
eqdistance seems not a pure predicate so cannot be used as a goal.

**Comment by cutepiggy123** [comment]
Hi, can you help me to explain this one "g = s_angle f c g 60, s_angle c f g -60 ? coll f d g". G is a auxiliary point or something? What is coll? And can you use dd to solve it now? Thank you for your help!!

**Comment by XiaoYang66** [comment]
> I think it should be `a b c = iso_triangle a b c ? cong a b a c` eqdistance seems not a pure predicate so cannot be used as a goal.

Hello, thank you very much, I replace eqdistance with the cong, it worked! That helps me a lot. I think the eqdistance may be related to the AR.

I am trying to figure all the rules out.

Some other scholars who are also interested with this repo have created a [Google Sheets](https://docs.google.com/document/d/1K4QspqnGFCJ9hpFyPUvaJXE7SAWbt3XRBBtyXzGhsOo/edit) to document the rule in [issue67](https://github.com/google-deepmind/alphageometry/issues/67)

**Comment by XiaoYang66** [comment]
> Hi, can you help me to explain this one "g = s_angle f c g 60, s_angle c f g -60 ? coll f d g". G is a auxiliary point or something? What is coll? And can you use dd to solve it now? Thank you for your help!!

Hello, I will try your question. If I figure it out, I will reply to you as soon as possible!

**Comment by XiaoYang66** [comment]
> Hi, can you help me to explain this one "g = s_angle f c g 60, s_angle c f g -60 ? coll f d g". G is a auxiliary point or something? What is coll? And can you use dd to solve it now? Thank you for your help!!

Hello, for now, I can ensure the coll means that a list of points on the same line.

I think the define of coll belongs to DD part, which is implemented and explained by this [paper](http://www.mmrc.iss.ac.cn/~xgao/paper/jar-gdbase.pdf)

For other symbol in your problem, I am trying to figure them out.

**Comment by SarthakRoongta** [comment]
Hi, I am trying to get it to prove a triangle is an isosceles triangle knowing that two sides are equal. I tried "a b c = triangle, cong a b a c ? eqanagle b a b c c b c a" but it gives me a keyerror on "cong"

**Comment by maokaiii** [comment]
> Hi, can you help me to explain this one "g = s_angle f c g 60, s_angle c f g -60 ? coll f d g". G is a auxiliary point or something? What is coll? And can you use dd to solve it now? Thank you for your help!!

here I am just constructing an equilateral triangle.
Coll is collinear

**Comment by maokaiii** [comment]
> Hi, I am trying to get it to prove a triangle is an isosceles triangle knowing that two sides are equal. I tried "a b c = triangle, cong a b a c ? eqanagle b a b c c b c a" but it gives me a keyerror on "cong"

I think that cong is used in goal only, which means you cannot make it before question mark

**Comment by Geomathart** [comment]
eqdistance is the command you need to use instead of cong in the construction.

**Comment by cutepiggy123** [comment]
> Other problem, again there is a problem of directed angle `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c a c b`
> 
> should be corrected as: `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c b c a`

Is alphageometry possible to solve this after you correct this?

**Comment by Geomathart** [comment]
Yeah, alphageometry has no problem to solve the corrected version.

**Comment by maokaiii** [comment]
> > Other problem, again there is a problem of directed angle `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c a c b`
> > should be corrected as: `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c b c a`
> 
> Is alphageometry possible to solve this after you correct this?

Of course, this much simpler problem is solved by alphageometry, which just use the fact of sum of internal angles of a triangle being 180 degree.

The main problem is in the very beginning.

**Comment by cutepiggy123** [comment]
Thanks guys!!! Do anyone know if alphageometry can use to find the length of a segment like find x, y of this example 
![image](https://github.com/google-deepmind/alphageometry/assets/166088836/497f735d-7d0f-48ef-95c0-a9db456fc109)

**Comment by XiaoYang66** [comment]
> hello, i find another problem that can not be solved by dd+ar or alphageometry `a b c d = eq_trapezoid a b c d ? cong c a b d`

I have found that the author has pointed that a similar problem taht can not be solved in line 452 in file jgex_file_ag_231.txt

**Comment by tpgh24** [comment]
The current version of AG indeed has many limitations. I made some improvements in a fork repository and have some ideas to improve it, check out [AG4Masses](https://github.com/tpgh24/ag4masses) and [issue 110](https://github.com/google-deepmind/alphageometry/issues/110)

**Comment by ShChen233** [comment]
> Other problem, again there is a problem of directed angle `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c a c b`
> 
> should be corrected as: `a b = segment a b; c = s_angle b a c -50, s_angle a b c 70; d = s_angle b a d 60 ? eqangle a b a d c b c a`

![image](https://github.com/user-attachments/assets/fae15b6c-95f4-4ca5-8886-b6e6e5c569ba)
however，the procedure is wrong
